compiler/cli.py

```diff
@@ -87,68 +87,71 @@ def make_command_line_parser():
         'directory at the start of execution.',
         type=bool_from_string,
         required=True)
+    parser.add_argument(
+        '--import_root',
+        help='Path to add to sys.path, may be repeated to provide multiple roots.',
+        action='append',
+        default=[],
+        dest='import_roots')
     return parser
 
 
 def parse_stub(stub_filename):
-    """Parse the imports and interpreter path from a py_binary() stub.
+    """Parse interpreter path from a py_binary() stub.
 
     We assume the stub is utf-8 encoded.
 
-    TODO(b/29227737): Remove this once we can access imports from skylark.
-
-    Returns (list of relative paths, path to Python interpreter)
+    Returns path to Python interpreter
     """
 
-    # Find the list of import roots
-    imports_regex = re.compile(r'''^  python_imports = '([^']*)'$''')
+    # Find the interpreter
     interpreter_regex = re.compile(r'''^PYTHON_BINARY = '([^']*)'$''')
-    import_roots = None
     interpreter = None
     with io.open(stub_filename, 'rt', encoding='utf8') as stub_file:
         for line in stub_file:
-            importers_match = imports_regex.match(line)
-            if importers_match:
-                import_roots = importers_match.group(1).split(':')
-                # Filter out empty paths
-                import_roots = [x for x in import_roots if x]
             interpreter_match = interpreter_regex.match(line)
             if interpreter_match:
                 interpreter = interpreter_match.group(1)
-    if import_roots is None or not interpreter:
+    if not interpreter:
         raise error.Error('Failed to parse stub file [%s]' % stub_filename)
 
     # Find the Python interpreter, matching the search logic in
```

Comment:
Suggest adding an explanatory comment:

> This somewhat mirrors the logic in python_stub_template.txt, but we don't support relative paths (i.e., in-workspace interpreters). This is because the interpreter will be used in the .par file's shebang, and putting a relative path in a shebang is extremely brittle and non-relocatable. (The reason the standard py_binary rule can use an in-workspace interpreter is that its stub script runs in a separate process and has a shebang referencing the system interpreter.) As a special case, if the Python target is using the autodetecting Python toolchain, which is technically an in-workspace runtime, we rewrite it to "/usr/bin/env python[2|3]" rather than fail.